examples/azure-sample-project/src/main/kotlin/project/Main.kt

```diff
@@ -10,7 +11,20 @@ import com.pulumi.random.kotlin.randomPassword
 
 fun main() {
     Pulumi.run { ctx ->
-        val resourceGroup = resourceGroup("azure-sample-project")
+        val provider = azureProvider("azure-provider") {
+            args {
+                skipProviderRegistration(true)
+            }
+        }
+
+        val resourceGroup = resourceGroup("azure-sample-project") {
+            args {
+                location(ctx.config("azure").get("location").orElse(null))
```

Comment:
This error began appearing in E2E tests:
```

@ Previewing update..........
 +  pulumi:pulumi:Stack azure-sample-project-test7887733202 create 
 +  pulumi:providers:azure azure-provider create 
@ Previewing update......
    azure:core:ResourceGroup azure-sample-project  error: azure:core/resourceGroup:ResourceGroup resource 'azure-sample-project' has a problem: Missing required argument. The argument "location" is required, but no definition was found.. Examine values at 'azure-sample-project.location'.
 +  pulumi:pulumi:Stack azure-sample-project-test7887733202 create 
    azure:core:ResourceGroup azure-sample-project  1 error
Diagnostics:
  azure:core:ResourceGroup (azure-sample-project):
    error: azure:core/resourceGroup:ResourceGroup resource 'azure-sample-project' has a problem: Missing required argument. The argument "location" is required, but no definition was found.. Examine values at 'azure-sample-project.location'.
```

We add `azure:location` into the configuration, but for some reason it's no longer used here implicitly
<img width="1169" alt="image" src="https://github.com/user-attachments/assets/d3e78dfd-88cd-4886-871b-299658885a54">

I'll create a task to investigate this in the future

Reply:
Might be a regression from this: https://github.com/pulumi/pulumi-azure/issues/176